core/src/test/java/org/apache/accumulo/core/data/KeyTest.java

```diff
@@ -79,6 +80,96 @@ public void testCopyData() {
     assertEquals(k1, k2);
 
   }
+  
+  @Test
+  public void testCopyDataWithByteArrayConstructors() {
+    byte[] row = "r".getBytes();
+    byte[] cf = "cf".getBytes();
+    byte[] cq = "cq".getBytes();
+    byte[] cv = "cv".getBytes();
+
+    Key kRow = new Key(row);
+    Key kRowcolFam = new Key(row, cf);
+    Key kRowcolFamColQual = new Key(row, cf, cq);
+    Key kRowcolFamColQualColVis = new Key(row, cf, cq, cv);
+    Key kRowcolFamColQualColVisTimeStamp = new Key(row, cf, cq, cv, 5L);
+    
+    // test row constructor
+    assertNotSameByteArray(kRow, row, cf, cq, cv, 5L);
+    
+    // test row, column family constructor
+    assertNotSameByteArray(kRowcolFam, row, cf, cq, cv, 5L);
+    
+    // test row, column family, column qualifier constructor
+    assertNotSameByteArray(kRowcolFamColQual, row, cf, cq, cv, 5L);
+    
+    // test row, column family, column qualifier, column visibility constructor
+    assertNotSameByteArray(kRowcolFamColQualColVis, row, cf, cq, cv, 5L);
+    
+    // test row, column family, column qualifier, column visibility, timestamp constructor
+    assertNotSameByteArray(kRowcolFamColQualColVisTimeStamp, row, cf, cq, cv, 5L);
+  }
+  
+  @Test
+  public void testCopyDataWithTextConstructors() {
+    Text rowText = new Text("r");
+    Text cfText = new Text("cf");
+    Text cqText = new Text("cq");
+    Text cvText = new Text("cv");
+    
+    byte[] row = "r".getBytes();
+    byte[] cf = "cf".getBytes();
+    byte[] cq = "cq".getBytes();
+    byte[] cv = "cv".getBytes();
+
+    //make Keys from Text parameters
+    Key kRow = new Key(rowText);
+    Key kRowcolFam = new Key(rowText, cfText);
+    Key kRowcolFamColQual = new Key(rowText, cfText, cqText);
+    Key kRowcolFamColQualColVis = new Key(rowText, cfText, cqText, cvText);
+    Key kRowcolFamColQualColVisTimeStamp = new Key(rowText, cfText, cqText, cvText, 5L);
+
+    // test row constructor
+    assertNotSameByteArray(kRow, row, cf, cq, cv, 5L);
```

Comment:
The byte arrays passed here are not used to construct the key.

Reply:
created empty array to pass to test method when not all values are used

Reply:
I don't think this test with Text accomplishes anything.  The following statement will create an internal byte array in Text.

```java
Text rowText = new Text("r");
```

The byte array in `rowText` is always a different array instance than what the following statement creates.

```java
 byte[] row = "r".getBytes();
```

Therefore the byte array in `rowText` and the byte array `row` will never be the same.  But the reason they will never be the same has nothing to do with Key.

Reply:
I changed to goal of the Text test to ensure it creates the correct byte array in the constructor.